src/widget/wtrackmenu.cpp

```diff
@@ -1595,6 +1625,161 @@ void WTrackMenu::slotClearAllMetadata() {
+                QDialogButtonBox::AcceptRole);
+        cancelBtn->setDefault(true);
+        // This is required after customizing the buttons, otherwise neither button
+        // would close the dialog.
+        connect(cancelBtn, &QPushButton::clicked, dlgDelConfirm, &QDialog::reject);
+        connect(deleteBtn, &QPushButton::clicked, dlgDelConfirm, &QDialog::accept);
+
+        delLayout->addWidget(delListWidget);
+        delLayout->addWidget(delWarning);
+        delLayout->addWidget(delButtons);
+        dlgDelConfirm->setLayout(delLayout);
+
+        if (dlgDelConfirm->exec() == QDialog::Rejected) {
+            return;
+        }
+    }
+
+    // Set up and initiate the track batch operation
+    const auto progressLabelText =
+            tr("Removing %1 track file(s) from disk...",
+                    "",
+                    getTrackCount());
+    const auto trackOperator =
+            RemoveTrackFilesFromDiskTrackPointerOperation();
+    applyTrackPointerOperation(
+            progressLabelText,
+            &trackOperator);
+
+    // Purge deleted tracks and show deletion summary message.
+    if (trackOperator.tr_tracksToPurge.length() > 0) {
+        // Purge only those tracks whose files were actually deleted.
+        m_pLibrary->trackCollectionManager()->purgeTracks(trackOperator.tr_tracksToPurge);
+        // Optional message box:
+        QMessageBox msgBoxPurgeTracks(QMessageBox::Information,
+                QObject::tr("Track files deleted"),
+                nullptr);
+        msgBoxPurgeTracks.setText(
+                tr("%1 track files were deleted from disk and purged "
+                   "from the Mixxx database.")
+                        .arg(QString::number(
+                                trackOperator.tr_tracksToPurge.length())) +
+                QString("<br><br>") +
+                tr("Note: if you are in Browse or Recording you need to "
+                   "click the current view again to see changes."));
+        msgBoxPurgeTracks.setTextFormat(Qt::RichText);
+        msgBoxPurgeTracks.exec();
+    }
+
+    if (trackOperator.tr_tracksToKeep.length() >= 0) {
+        return;
+    }
+
+    {
+        // If there are tracks that could not be deleted show a message with those listed.
+        QDialog* dlgNotDeleted = new QDialog(nullptr);
+        dlgNotDeleted->setWindowTitle(tr("Delete Track Files"));
+        QVBoxLayout* notDeletedLayout = new QVBoxLayout;
+        QLabel* notDeletedLabel = new QLabel;
+        notDeletedLabel->setText(
+                tr("The following %1 files could not be deleted from disk")
+                        .arg(QString::number(
+                                trackOperator.tr_tracksToKeep.length())));
+        notDeletedLabel->setTextFormat(Qt::RichText);
+
+        QListWidget* notDeletedListWidget = new QListWidget;
+        notDeletedListWidget->addItems(trackOperator.tr_tracksToKeep);
+        notDeletedListWidget->setMinimumSize(
```

Comment:
How about encapsulating this in a function named `void mixxx::widgethelper::adjustMinimumListWidgetSize(...)` (see comment about function naming)

Reply:
good idea.
though I assumed from other code that it's most transparent to let a function fetch a value/object, then apply it 'in place'.

Reply:
done.